compiler/dyno/lib/resolution/Resolver.cpp

```diff
@@ -2410,6 +2410,11 @@ static QualifiedType resolveSerialIterType(Resolver& resolver,
   iterand->traverse(resolver);
   ResolvedExpression& iterandRE = resolver.byPostorder.byAst(iterand);
 
+  if (resolver.scopeResolveOnly) {
```

Comment:
How come we're returning an error here? Looks like it's the "regular path" to go into resolving an indexable loop, then into `resolveSerialIterType`; for `param` loops, we just return early (line 2472), which I assume leaves the type of the index unknown, but here, we explicitly make it erroneous.

Reply:
Ah, that's because I blindly copied the places where we set ``idxType`` without looking to closely. I'll fix it to just use UnknownType.